config/config.py

```diff
@@ -0,0 +1,187 @@
+import yaml, os
+
+class _DeskConfigDAO:
+	"""
+	Desk Config DAO
+	"""
+
+	KEY_ID = "id"
+	KEY_NAME = "name"
+	KEY_SENSOR_PIN = "sensor_pin"
+
+	def __init__(self, desk):
+		self.__desk_id = desk[self.KEY_ID]
+		self.__desk_name = desk[self.KEY_NAME]
+		self.__desk_pin = desk[self.KEY_SENSOR_PIN]
+
+	@property
+	def id(self):
+		return self.__desk_id
+
+	@property
+	def name(self):
+		return self.__desk_name
+
+	@property
+	def pin(self):
+		return self.__desk_pin
+
+
+class _DebugConfigDAO:
+	"""
+	Debug Config DAO
+	"""
+
+	KEY_ENABLED = "enabled"
+	KEY_REMOTE_SIMULATOR_CONFIG = "remote-simulator"
+
+	def __init__(self, debug):
+		self.__enabled = debug[self.KEY_ENABLED]
+		self.__remote_simulator_config = _RemoteSimulatorConfigDAO(debug[self.KEY_REMOTE_SIMULATOR_CONFIG])
+
+	@property
+	def enabled(self):
+		return self.__enabled
+
+	@property
+	def simulator_config(self):
+		return self.__remote_simulator_config
+
+
+class _RemoteSimulatorConfigDAO:
+	"""
+	Simulator Config DAO
+	"""
+	KEY_ENABLED = "enabled"
+	KEY_PORT = "port"
+
+	def __init__(self, simulator):
+		self.__enabled = simulator[self.KEY_ENABLED]
+		self.__port = simulator[self.KEY_PORT]
+
+	@property
+	def enabled(self):
+		return self.__enabled
+
+	@property
+	def port(self):
+		return self.__port
+
+
+class _ConfigDAO:
+	"""
+	Data Access Object (DAO) for the configuration
+	"""
+	KEY_DEBUG_CONFIG = "debug"
+	KEY_DESK_CONFIGS = "desks"
+
+	def __init__(self, loader):
+		self.__loader = loader
+		self.__properties = loader.properties
+
+		self.__desks = []
+		self.__debug_config = _DebugConfigDAO(self.__properties[self.KEY_DEBUG_CONFIG])
+
+		for desk in self.__properties[self.KEY_DESK_CONFIGS]:
+			self.__desks.append(_DeskConfigDAO(desk))
+
+	@property
+	def debug_config(self):
+		return self.__debug_config
+
+	@property
+	def desks(self):
+		return self.__desks
+
+	def __str__(self):
+		return self.__properties.__str__()
+
+	def update_desk_list(self, new_desk):
+		updated_desks = self.__loader.update_desk_list(new_desk)
+		self.__desks = list(map(lambda desk_dict: _DeskConfigDAO(desk_dict), updated_desks))
```

Comment:
By doing this, we make sure that we add it to `__desks` after updating `config.yaml` was successful

Reply:
List comprehensions are preferred in python over `map`. Hence we could do:

```
self.__desks = [ _DeskConfigDAO(desk_dict) for desk_dict in updated_desks ]
```